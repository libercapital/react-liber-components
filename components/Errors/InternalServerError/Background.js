"use strict";

require("core-js/modules/es6.object.define-property");

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports["default"] = void 0;

var _react = _interopRequireDefault(require("react"));

var _propTypes = _interopRequireDefault(require("prop-types"));

var _styledComponents = _interopRequireDefault(require("styled-components"));

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { "default": obj }; }

var Cls1 = _styledComponents["default"].rect.withConfig({
  displayName: "Background__Cls1",
  componentId: "moojgn-0"
})(["fill:#f8feff;"]);

var Cls2 = _styledComponents["default"].path.withConfig({
  displayName: "Background__Cls2",
  componentId: "moojgn-1"
})(["fill:#eefaff;"]);

var Cls3 = _styledComponents["default"].path.withConfig({
  displayName: "Background__Cls3",
  componentId: "moojgn-2"
})(["fill:#fff;"]);

var Background = function Background(_ref) {
  var className = _ref.className;
  return _react["default"].createElement("svg", {
    preserveAspectRatio: "none",
    xmlns: "http://www.w3.org/2000/svg",
    viewBox: "0 0 1366 768",
    className: className
  }, _react["default"].createElement("g", {
    id: "Fundo_500",
    "data-name": "Fundo 500",
    transform: "translate(0 -815)"
  }, _react["default"].createElement(Cls1, {
    id: "Tela",
    width: "1366",
    height: "768",
    transform: "translate(0 815)"
  }), _react["default"].createElement("g", {
    id: "Group_21",
    "data-name": "Group 21"
  }, _react["default"].createElement(Cls2, {
    id: "Path_126",
    "data-name": "Path 126",
    d: "M1252.518,1201.6,1029.315,815H114.627L0,1013.54v376.116L111.627,1583h920.688Z"
  })), _react["default"].createElement(Cls3, {
    id: "Path_127",
    "data-name": "Path 127",
    d: "M970.227,1583,1190.3,1201.831,966.958,815H158.41L0,1089.374v224.916L155.14,1583Z"
  })));
};

Background.propTypes = {
  className: _propTypes["default"].string
};
Background.defaultProps = {
  className: ''
};
var _default = Background;
exports["default"] = _default;