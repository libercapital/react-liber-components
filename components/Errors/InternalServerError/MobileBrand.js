"use strict";

require("core-js/modules/es6.object.define-property");

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports["default"] = void 0;

var _react = _interopRequireDefault(require("react"));

var _propTypes = _interopRequireDefault(require("prop-types"));

var _styledComponents = _interopRequireDefault(require("styled-components"));

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { "default": obj }; }

var Cls1 = _styledComponents["default"].path.withConfig({
  displayName: "MobileBrand__Cls1",
  componentId: "gft0m1-0"
})(["fill:#009dff;"]);

var Cls2 = _styledComponents["default"].path.withConfig({
  displayName: "MobileBrand__Cls2",
  componentId: "gft0m1-1"
})(["fill:none;stroke:#009dff;stroke-miterlimit:10;stroke-width:2px;"]);

var MobileBrand = function MobileBrand(_ref) {
  var className = _ref.className;
  return _react["default"].createElement("svg", {
    className: className,
    xmlns: "http://www.w3.org/2000/svg",
    viewBox: "0 0 196.52 93.44"
  }, _react["default"].createElement("title", null, "500"), _react["default"].createElement("g", {
    id: "Camada_2",
    "data-name": "Camada 2"
  }, _react["default"].createElement("g", {
    id: "_500_Mobile",
    "data-name": "500 Mobile"
  }, _react["default"].createElement("g", {
    id: "_500",
    "data-name": "500"
  }, _react["default"].createElement(Cls1, {
    d: "M2.19,45.93l5-43.75H55.4V16.43H21.17L19,34.94A27,27,0,0,1,32,31.68q12.3,0,19.28,7.63t7,21.34a31.14, 31.14,0,0,1-3.52,14.93A24.81,24.81,0,0,1,44.64,85.81a31.53,31.53,0,0,1-15.49,3.63,33.41,33.41,0,0, 1-14.49-3.16,25.66,25.66,0,0,1-10.55-8.9A24.25,24.25,0,0,1,0,64.32H16.91a12.57,12.57,0,0,0,3.75, 8.36,11.92,11.92,0,0,0,8.43,3A10.49,10.49,0,0,0,38,71.5q3.14-4.17,3.14-11.79, 0-7.34-3.61-11.24t-10.23-3.9a14.82,14.82,0,0,0-9.87,3.19L15.78,49.3Z"
  }), _react["default"].createElement(Cls2, {
    d: "M127.05,52.67q0,17.86-7.39,27.32T98,89.44Q84,89.44,76.5,80.16t-7.63-26.6V37.71q0-18,7.48-27.37T97.9, 1Q112,1,119.42,10.25t7.63,26.58ZM110,35.25q0-10.71-2.92-15.59T97.9,14.78q-6,0-8.9,4.64T86,34V54.9q0, 10.53,2.87,15.65T98,75.67q6.27,0,9-4.91t2.89-15Z"
  }), _react["default"].createElement(Cls1, {
    d: "M196.52,52.67q0,17.86-7.39,27.32t-21.64,9.45q-14.07,0-21.52-9.28t-7.63-26.6V37.71q0-18, 7.48-27.37T167.37,1q14.07,0,21.52,9.25t7.63,26.58ZM179.43,35.25q0-10.71-2.93-15.59t-9.13-4.88q-6,0-8.9, 4.64t-3,14.53V54.9q0,10.53,2.86,15.65t9.2,5.12q6.25,0,9-4.91t2.9-15Z"
  }), _react["default"].createElement(Cls1, {
    d: "M131.05,56.67q0,17.86-7.39,27.32T102,93.44Q88,93.44,80.5,84.16t-7.63-26.6V41.71q0-18,7.48-27.37T101.9, 5Q116,5,123.42,14.25t7.63,26.58ZM114,39.25q0-10.71-2.92-15.59t-9.14-4.88q-6,0-8.9,4.64T90,38V58.9q0, 10.53,2.87,15.65T102,79.67q6.27,0,9-4.91t2.89-15Z"
  })))));
};

MobileBrand.propTypes = {
  className: _propTypes["default"].string
};
MobileBrand.defaultProps = {
  className: ''
};
var _default = MobileBrand;
exports["default"] = _default;