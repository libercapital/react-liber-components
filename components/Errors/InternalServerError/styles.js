"use strict";

require("core-js/modules/es6.object.define-property");

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.MobileMessageBox = exports.MobileBrand = exports.MobileContent = exports.MobileIllustration = exports.DesktopMessageBox = exports.DesktopContent = exports.DesktopIllustration = exports.DesktopBackground = void 0;

require("core-js/modules/es6.object.define-properties");

require("core-js/modules/es6.object.freeze");

var _styledComponents = _interopRequireDefault(require("styled-components"));

var _Util = require("../../Util");

var _Background = _interopRequireDefault(require("./Background"));

var _DesktopIllustration = _interopRequireDefault(require("./DesktopIllustration"));

var _MobileIllustration = _interopRequireDefault(require("./MobileIllustration"));

var _MobileBrand = _interopRequireDefault(require("./MobileBrand"));

var _styles = require("../styles");

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { "default": obj }; }

function _templateObject5() {
  var data = _taggedTemplateLiteral(["\n    max-width: 225px;\n    & h1 {\n      font-size: 30px;\n      margin-bottom: 32px;\n    }\n    & h3 {\n      font-size: 24px;\n      margin-top: unset;\n      margin-bottom: unset;\n    }\n    & a {\n      font-size: 18px;\n    }\n  "]);

  _templateObject5 = function _templateObject5() {
    return data;
  };

  return data;
}

function _templateObject4() {
  var data = _taggedTemplateLiteral(["\n    width: 45vh;\n  "]);

  _templateObject4 = function _templateObject4() {
    return data;
  };

  return data;
}

function _templateObject3() {
  var data = _taggedTemplateLiteral(["\n    max-width: 15vw;\n    & h1 { font-size: 2.15vw; }\n    & h3 { font-size: 1.4vw; }\n  "]);

  _templateObject3 = function _templateObject3() {
    return data;
  };

  return data;
}

function _templateObject2() {
  var data = _taggedTemplateLiteral(["\n    top: 29vh;\n    left: 57vw;\n  "]);

  _templateObject2 = function _templateObject2() {
    return data;
  };

  return data;
}

function _templateObject() {
  var data = _taggedTemplateLiteral(["\n    top: 20vh;\n    left: 15vw;\n    width: 40vw;\n  "]);

  _templateObject = function _templateObject() {
    return data;
  };

  return data;
}

function _taggedTemplateLiteral(strings, raw) { if (!raw) { raw = strings.slice(0); } return Object.freeze(Object.defineProperties(strings, { raw: { value: Object.freeze(raw) } })); }

var DesktopBackground = (0, _styledComponents["default"])(_Background["default"]).withConfig({
  displayName: "styles__DesktopBackground",
  componentId: "sc-8r0jx-0"
})(["", ""], _styles.DesktopBackgroundStyles);
exports.DesktopBackground = DesktopBackground;
var DesktopIllustration = (0, _styledComponents["default"])(_DesktopIllustration["default"]).withConfig({
  displayName: "styles__DesktopIllustration",
  componentId: "sc-8r0jx-1"
})(["", " top:20vh;left:13vw;width:42.5vw;", ""], _styles.DesktopIllustrationStyles, _Util.media.extraLarge(_templateObject()));
exports.DesktopIllustration = DesktopIllustration;

var DesktopContent = _styledComponents["default"].div.withConfig({
  displayName: "styles__DesktopContent",
  componentId: "sc-8r0jx-2"
})(["", " top:25vh;left:57vw;", ""], _styles.DesktopContentStyles, _Util.media.extraLarge(_templateObject2()));

exports.DesktopContent = DesktopContent;

var DesktopMessageBox = _styledComponents["default"].div.withConfig({
  displayName: "styles__DesktopMessageBox",
  componentId: "sc-8r0jx-3"
})(["", " max-width:20vw;& h1{font-size:2.75vw;margin-top:0.3vw;margin-bottom:1.9vw;}& h3{font-size:1.9vw;margin-top:unset;margin-bottom:unset;}", ""], _styles.DesktopMessageBoxStyles, _Util.media.extraLarge(_templateObject3()));

exports.DesktopMessageBox = DesktopMessageBox;
var MobileIllustration = (0, _styledComponents["default"])(_MobileIllustration["default"]).withConfig({
  displayName: "styles__MobileIllustration",
  componentId: "sc-8r0jx-4"
})(["", " left:0;bottom:0;width:40vh;", ""], _styles.MobileIllustrationStyles, _Util.media.medium(_templateObject4()));
exports.MobileIllustration = MobileIllustration;

var MobileContent = _styledComponents["default"].div.withConfig({
  displayName: "styles__MobileContent",
  componentId: "sc-8r0jx-5"
})(["", ""], _styles.MobileContentStyles);

exports.MobileContent = MobileContent;
var MobileBrand = (0, _styledComponents["default"])(_MobileBrand["default"]).withConfig({
  displayName: "styles__MobileBrand",
  componentId: "sc-8r0jx-6"
})(["", ""], _styles.MobileBrandStyles);
exports.MobileBrand = MobileBrand;
var MobileMessageBox = (0, _styledComponents["default"])(DesktopMessageBox).withConfig({
  displayName: "styles__MobileMessageBox",
  componentId: "sc-8r0jx-7"
})(["", " margin-top:25px;& h1{font-size:26px;margin-bottom:20px;}& h3{font-size:20px;margin-top:unset;margin-bottom:unset;}& a{font-size:14px;padding:5px 28px;}", ""], _styles.MobileMessageBoxStyles, _Util.media.medium(_templateObject5()));
exports.MobileMessageBox = MobileMessageBox;