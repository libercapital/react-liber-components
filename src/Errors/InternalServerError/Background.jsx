import React from 'react';
import PropTypes from 'prop-types';
import styled from 'styled-components';

const Cls1 = styled.rect`
  fill: #f8feff;
`;

const Cls2 = styled.path`
  fill: #eefaff;
`;

const Cls3 = styled.path`
  fill: #fff;
`;

const Background = ({ className }) => (
  <svg preserveAspectRatio="none" xmlns="http://www.w3.org/2000/svg" viewBox="0 0 1366 768" className={className}>
    <g id="Fundo_500" data-name="Fundo 500" transform="translate(0 -815)">
      <Cls1 id="Tela" width="1366" height="768" transform="translate(0 815)" />
      <g id="Group_21" data-name="Group 21">
        <Cls2
          id="Path_126"
          data-name="Path 126"
          d="M1252.518,1201.6,1029.315,815H114.627L0,1013.54v376.116L111.627,1583h920.688Z"
        />
      </g>
      <Cls3
        id="Path_127"
        data-name="Path 127"
        d="M970.227,1583,1190.3,1201.831,966.958,815H158.41L0,1089.374v224.916L155.14,1583Z"
      />
    </g>
  </svg>

);

Background.propTypes = {
  className: PropTypes.string,
};

Background.defaultProps = {
  className: '',
};

export default Background;
