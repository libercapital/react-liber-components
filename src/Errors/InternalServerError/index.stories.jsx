import React from 'react';
import InternalServerError from './InternalServerError';

export default {
  title: 'Components 1.0|Errors/Internal Error',
  parameters: {
    jest: ['InternalServerError'],
  },
};

export const defaultStory = () => <InternalServerError />;

defaultStory.story = {
  name: 'default',
};
