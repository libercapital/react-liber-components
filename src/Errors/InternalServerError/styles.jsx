import styled from 'styled-components';
import { media } from '../../Util';
import Background from './Background';
import DesktopIllustrationRaw from './DesktopIllustration';
import MobileIllustrationRaw from './MobileIllustration';
import MobileBrandRaw from './MobileBrand';

import {
  DesktopBackgroundStyles,
  DesktopIllustrationStyles,
  DesktopContentStyles,
  DesktopMessageBoxStyles,
  MobileIllustrationStyles,
  MobileContentStyles,
  MobileBrandStyles,
  MobileMessageBoxStyles,
} from '../styles';

export const DesktopBackground = styled(Background)`
  ${DesktopBackgroundStyles}
`;

export const DesktopIllustration = styled(DesktopIllustrationRaw)`
  ${DesktopIllustrationStyles}
  top: 20vh;
  left: 13vw;
  width: 42.5vw;

  ${media.extraLarge`
    top: 20vh;
    left: 15vw;
    width: 40vw;
  `}
`;

export const DesktopContent = styled.div`
  ${DesktopContentStyles}
  top: 25vh;
  left: 57vw;
  ${media.extraLarge`
    top: 29vh;
    left: 57vw;
  `}
`;

export const DesktopMessageBox = styled.div`
  ${DesktopMessageBoxStyles}
  max-width: 20vw;

  & h1 {
    font-size: 2.75vw;
    margin-top: 0.3vw;
    margin-bottom: 1.9vw;
  }
  & h3 {
    font-size: 1.9vw;
    margin-top: unset;
    margin-bottom: unset;
  }
  ${media.extraLarge`
    max-width: 15vw;
    & h1 { font-size: 2.15vw; }
    & h3 { font-size: 1.4vw; }
  `}
`;

export const MobileIllustration = styled(MobileIllustrationRaw)`
  ${MobileIllustrationStyles}
  left: 0;
  bottom: 0;
  width: 40vh;

  ${media.medium`
    width: 45vh;
  `}
`;

export const MobileContent = styled.div`
  ${MobileContentStyles}
`;

export const MobileBrand = styled(MobileBrandRaw)`
  ${MobileBrandStyles}
`;

export const MobileMessageBox = styled(DesktopMessageBox)`
  ${MobileMessageBoxStyles}
  margin-top: 25px;

  & h1 {
    font-size: 26px;
    margin-bottom: 20px;
  }
  & h3 {
    font-size: 20px;
    margin-top: unset;
    margin-bottom: unset;
  }
  & a {
    font-size: 14px;
    padding: 5px 28px;
  }
  ${media.medium`
    max-width: 225px;
    & h1 {
      font-size: 30px;
      margin-bottom: 32px;
    }
    & h3 {
      font-size: 24px;
      margin-top: unset;
      margin-bottom: unset;
    }
    & a {
      font-size: 18px;
    }
  `}
`;
