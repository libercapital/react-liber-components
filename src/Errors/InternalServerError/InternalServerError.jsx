import React, { Fragment } from 'react';
import { DesktopContainer, MobileContainer, MobileBackground } from '../styles';
import {
  DesktopBackground,
  DesktopIllustration,
  DesktopContent,
  DesktopMessageBox,
  MobileIllustration,
  MobileContent,
  MobileBrand,
  MobileMessageBox,
} from './styles';

const InternalServerError = () => (
  <Fragment>
    <DesktopContainer>
      <DesktopBackground />
      <DesktopIllustration />
      <DesktopContent>
        <DesktopMessageBox>
          <h3>Desculpe, tivemos um problema interno...</h3>
          <h1>Mas já estamos arrumando tudo!</h1>
          <a href="/">Voltar à página inicial</a>
        </DesktopMessageBox>
      </DesktopContent>
    </DesktopContainer>
    <MobileContainer>
      <MobileBackground />
      <MobileIllustration />
      <MobileContent>
        <MobileBrand />
        <MobileMessageBox>
          <h3>Desculpe, tivemos um problema interno...</h3>
          <h1>Mas já estamos arrumando tudo!</h1>
          <a href="/">Voltar à página inicial</a>
        </MobileMessageBox>
      </MobileContent>
    </MobileContainer>
  </Fragment>
);

export default InternalServerError;
