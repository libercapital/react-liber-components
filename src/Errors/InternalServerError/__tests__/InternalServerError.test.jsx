import React from 'react';
import { shallow, configure } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
import toJson from 'enzyme-to-json';
import 'jest-styled-components';

import InternalServerError from '../InternalServerError';

describe('Testing InternalServerError', () => {
  configure({ adapter: new Adapter() });
  it('should render correctly', () => {
    const wrapper = shallow(<InternalServerError />);
    expect(toJson(wrapper)).toMatchSnapshot();
  });
});
