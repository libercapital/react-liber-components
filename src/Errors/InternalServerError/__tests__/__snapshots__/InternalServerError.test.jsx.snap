// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Testing InternalServerError should render correctly 1`] = `
<Fragment>
  <styles__DesktopContainer>
    <styles__DesktopBackground />
    <styles__DesktopIllustration />
    <styles__DesktopContent>
      <styles__DesktopMessageBox>
        <h3>
          Desculpe, tivemos um problema interno...
        </h3>
        <h1>
          Mas já estamos arrumando tudo!
        </h1>
        <a
          href="/"
        >
          Voltar à página inicial
        </a>
      </styles__DesktopMessageBox>
    </styles__DesktopContent>
  </styles__DesktopContainer>
  <styles__MobileContainer>
    <styles__MobileBackground />
    <styles__MobileIllustration />
    <styles__MobileContent>
      <styles__MobileBrand />
      <styles__MobileMessageBox>
        <h3>
          Desculpe, tivemos um problema interno...
        </h3>
        <h1>
          Mas já estamos arrumando tudo!
        </h1>
        <a
          href="/"
        >
          Voltar à página inicial
        </a>
      </styles__MobileMessageBox>
    </styles__MobileContent>
  </styles__MobileContainer>
</Fragment>
`;
